mlos_bench/mlos_bench/storage/sql/trial.py

```diff
@@ -80,21 +81,41 @@ def update(self, status: Status, timestamp: datetime,
                 raise
         return metrics
 
-    def update_telemetry(self, status: Status, metrics: List[Tuple[datetime, str, Any]]) -> None:
-        super().update_telemetry(status, metrics)
+    def update_telemetry(self, status: Status, timestamp: datetime,
+                         metrics: List[Tuple[datetime, str, Any]]) -> None:
+        super().update_telemetry(status, timestamp, metrics)
         # NOTE: Not every SQLAlchemy dialect supports `Insert.on_conflict_do_nothing()`
         # and we need to keep `.update_telemetry()` idempotent; hence a loop instead of
         # a bulk upsert.
         # See Also: comments in <https://github.com/microsoft/MLOS/pull/466>
-        for (timestamp, key, val) in metrics:
+        with self._engine.begin() as conn:
+            self._update_status(conn, status, timestamp)
+        for (metric_ts, metric_id, metric_value) in metrics:
             with self._engine.begin() as conn:
                 try:
                     conn.execute(self._schema.trial_telemetry.insert().values(
                         exp_id=self._experiment_id,
                         trial_id=self._trial_id,
-                        ts=timestamp,
-                        metric_id=key,
-                        metric_value=None if val is None else str(val),
+                        ts=metric_ts,
+                        metric_id=metric_id,
+                        metric_value=None if metric_value is None else str(metric_value),
                     ))
                 except IntegrityError as ex:
-                    _LOG.warning("Record already exists: %s :: %s", (timestamp, key, val), ex)
+                    _LOG.warning("Record already exists: %s :: %s",
+                                 (metric_ts, metric_id, metric_value), ex)
+
+    def _update_status(self, conn: Connection, status: Status, timestamp: datetime) -> None:
+        """
+        Insert a new status record into the database.
+        This call is idempotent.
+        """
+        try:
+            conn.execute(self._schema.trial_status.insert().values(
```

Comment:
Not seeing any tests for this new data source.
Also, what about retrieval?

Reply:
this will be in the next PR - removing it from this one